Error streaming Live TV - (exoplayer fails to play AAC audio)

### Describe the bug

1. Navigate to a Live TV channel in the app
2. Click "Tune to Channel"
3. Player shows up, but fails to play. A toast appears with the message "Error streaming live TV. Will re-try..."
4. The player keeps retrying the stream indefinitely

Only happens with ExoPlayer, libVLC plays the channel fine.

As a side note, this retry is so aggressive, that it is very hard to actually exit the player. You have to mash the back button many times, as once the player closes and you're back at the channel details view, the player is launched again.

This seems to be the relevant error from logcat:

    03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:   Caused by: com.google.android.exoplayer2.ParserException: Missing CodecPrivate for codec A_AAC

Not entirely sure what "CodecPrivate" is, but some points that could be of interest:

* Other files with AAC audio (movies, shows) play fine
* If I download a single 2-second TS segment of this Live TV channel and copy it into the "movies" library, it plays fine
* The issue *does not* occur in 0.13.0-beta.4 (re-confirmed by installing the Debug version)



### Logs

```shell
--------- beginning of system
03-11 03:26:22.536  5767  5767 W ViewRootImpl[MainActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_DPAD_CENTER, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=39491, downTime=39406, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 03:26:22.538  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 03:26:22.538  5767  5767 W ViewRootImpl[MainActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_DPAD_CENTER, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=39491, downTime=39406, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 03:39:04.862  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=801819, downTime=801777, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 03:39:04.866  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 03:39:04.867  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=801819, downTime=801777, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 03:39:11.894  5767  5767 W ViewRootImpl[FullDetailsActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=808793, downTime=808741, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 03:39:12.064  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 03:39:12.064  5767  5767 W ViewRootImpl[FullDetailsActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=808793, downTime=808741, deviceId=-1, source=0x2000001, displayId=-1 }
--------- beginning of main
03-11 23:40:28.200  5767 27569 I KtorClient: GET http://calor:8096/Shows/NextUp?userId=2d175700-4ca2-4c6a-8a54-241cf2bb2390&limit=10&fields=DateCreated&imageTypeLimit=1&enableTotalRecordCount=true&disableFirstEpisode=false
03-11 23:40:28.205  5767 26307 I KtorClient: GET http://calor:8096/Users/2d175700-4ca2-4c6a-8a54-241cf2bb2390/Items/Resume?limit=10&fields=DateCreated&mediaTypes=Video&imageTypeLimit=1&enableTotalRecordCount=true&enableImages=true
03-11 23:40:28.339  5767  5796 I lyfin.androidt: Waiting for a blocking GC ProfileSaver
03-11 23:40:28.392  5767  5785 W lyfin.androidt: Reducing the number of considered missed Gc histogram windows from 896 to 100
03-11 23:40:28.392  5767  5796 I lyfin.androidt: WaitForGcToComplete blocked ProfileSaver on ProfileSaver for 53.573ms
03-11 23:40:28.439  5767 27569 I KtorClient: GET http://calor:8096/Users/2d175700-4ca2-4c6a-8a54-241cf2bb2390/Views?includeHidden=false
03-11 23:40:28.592  5767  5979 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=7fc6418a-fde8-4930-a3af-2f8baa1cb636, tags={ org.jellyfin.androidtv.integration.LeanbackChannelWorker } ]
03-11 23:45:55.640  5767  5767 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@ecda906
03-11 23:45:55.691  5767  5767 I AppThemeCallbacks: Applying theme: DARK
03-11 23:45:55.694  5767  5767 I AuthenticatedUserCallbacks: Activity org.jellyfin.androidtv.ui.startup.StartupActivity is ignored
03-11 23:45:55.712  5767  5767 D LegacyAccountMigration: Skipping migration of legacy credentials from /data/user/0/org.jellyfin.androidtv/files/org.jellyfin.androidtv.login.json (file does not exist)
03-11 23:45:55.738  5767  5767 I JellyfinApplication$onCreate: Process lifecycle started
03-11 23:45:55.739  5767  5767 D SessionRepositoryImpl: Restoring default session
03-11 23:45:55.751  5767  5767 I StartupActivity: Found a session in the session repository, waiting for the currentUser in the application class.
03-11 23:45:55.754  5767  5767 I StartupActivity: CurrentUser changed to 2d1757004ca24c6a8a54241cf2bb2390 while waiting for startup.
03-11 23:45:55.756  5767  5767 D StartupActivity: Determining next activity (action=android.intent.action.MAIN, itemId=null, itemIsUserView=false)
03-11 23:45:55.877  5767  5767 D StartupActivity: Opening next activity Intent { flg=0x10004000 cmp=org.jellyfin.androidtv/.ui.browsing.MainActivity }
03-11 23:45:55.881  5767 27569 I KtorClient: GET http://calor:8096/Playback/BitrateTest?size=4000000
03-11 23:45:55.901  5767  5767 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@2440dbe
03-11 23:45:55.911  5767  5767 I AppThemeCallbacks: Applying theme: DARK
03-11 23:45:55.926  5767  5767 I BackgroundService: Adding BackgroundServiceFragment to activity
03-11 23:45:55.948  5767  5767 I BackgroundService: Adding BackgroundServiceFragment to activity
03-11 23:45:55.949  5767 27569 D jellyfin-apiclient: Adding request to queue: http://calor:8096/LiveTv/Programs/Recommended?UserId=2d1757004ca24c6a8a54241cf2bb2390&EnableTotalRecordCount=false&IsAiring=true&Limit=1&ImageTypeLimit=1&format=json
03-11 23:45:55.951  5767  5767 D MediaManager: Added event listener.  Total listeners: 1
03-11 23:45:55.981  5767  5767 I HomeFragment: Updating audio queue in HomeFragment (onResume)
03-11 23:45:55.982  5767  5767 D BackgroundServiceFragment: Restoring background drawable
03-11 23:45:55.983  5767  5767 D BackgroundServiceFragment: Restoring background drawable
03-11 23:45:55.995  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/LiveTv/Programs/Recommended?UserId=2d1757004ca24c6a8a54241cf2bb2390&EnableTotalRecordCount=false&IsAiring=true&Limit=1&ImageTypeLimit=1&format=json
03-11 23:45:55.998  5767 28359 I KtorClient: GET http://calor:8096/Users/2d175700-4ca2-4c6a-8a54-241cf2bb2390/Items/Resume?limit=10&fields=DateCreated&mediaTypes=Video&imageTypeLimit=1&enableTotalRecordCount=true&enableImages=true
03-11 23:45:56.008  5767 27569 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Views?format=json
03-11 23:45:56.013  5767 28358 I KtorClient: GET http://calor:8096/Shows/NextUp?userId=2d175700-4ca2-4c6a-8a54-241cf2bb2390&limit=10&fields=DateCreated&imageTypeLimit=1&enableTotalRecordCount=true&disableFirstEpisode=false
03-11 23:45:56.141  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Views?format=json
03-11 23:45:56.164  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Views?format=json
03-11 23:45:56.166  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items?SortBy=DatePlayed&EnableTotalRecordCount=false&ExcludeLocationTypes=Virtual&MediaTypes=Video&ImageTypeLimit=1&recursive=true&Filters=IsResumable&Fields=PrimaryImageAspectRatio%2COverview%2CItemCounts%2CDisplayPreferencesId%2CChildCount&sortOrder=Descending&Limit=50&format=json
03-11 23:45:56.167  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items?SortBy=DatePlayed&EnableTotalRecordCount=false&ExcludeLocationTypes=Virtual&MediaTypes=Audio&ImageTypeLimit=1&recursive=true&Filters=IsResumable&Fields=PrimaryImageAspectRatio%2COverview%2CItemCounts%2CDisplayPreferencesId%2CChildCount&sortOrder=Descending&Limit=50&format=json
03-11 23:45:56.170  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/LiveTv/Programs/Recommended?Fields=Overview%2CPrimaryImageAspectRatio%2CChannelInfo%2CChildCount&UserId=2d1757004ca24c6a8a54241cf2bb2390&EnableTotalRecordCount=false&IsAiring=true&Limit=20&ImageTypeLimit=1&format=json
03-11 23:45:56.172  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Shows/NextUp?Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&UserId=2d1757004ca24c6a8a54241cf2bb2390&Limit=50&ImageTypeLimit=1&format=json
03-11 23:45:56.178  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items/Latest?ParentId=f137a2dd21bbc1b99aa5c0f6bf02a805&Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&Limit=50&GroupItems=true&format=json
03-11 23:45:56.179  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items/Latest?ParentId=79a2726d3c50e769a8af1e4184e4fccf&Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&Limit=50&GroupItems=true&format=json
03-11 23:45:56.180  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items/Latest?ParentId=a656b907eb3a73532e40e44b968d0225&Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&Limit=50&GroupItems=true&format=json
03-11 23:45:56.314  5767 28358 I KtorClient: GET http://calor:8096/Users/2d175700-4ca2-4c6a-8a54-241cf2bb2390/Views?includeHidden=false
03-11 23:45:56.398  5767  5785 I lyfin.androidt: Background young concurrent copying GC freed 71155(3097KB) AllocSpace objects, 5(108KB) LOS objects, 13% free, 16MB/19MB, paused 398us total 137.627ms
03-11 23:45:56.417  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items?SortBy=DatePlayed&EnableTotalRecordCount=false&ExcludeLocationTypes=Virtual&MediaTypes=Audio&ImageTypeLimit=1&recursive=true&Filters=IsResumable&Fields=PrimaryImageAspectRatio%2COverview%2CItemCounts%2CDisplayPreferencesId%2CChildCount&sortOrder=Descending&Limit=50&format=json
03-11 23:45:56.417  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Views?format=json
03-11 23:45:56.422  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/LiveTv/Programs/Recommended?Fields=Overview%2CPrimaryImageAspectRatio%2CChannelInfo%2CChildCount&UserId=2d1757004ca24c6a8a54241cf2bb2390&EnableTotalRecordCount=false&IsAiring=true&Limit=20&ImageTypeLimit=1&format=json
03-11 23:45:56.426  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items?SortBy=DatePlayed&EnableTotalRecordCount=false&ExcludeLocationTypes=Virtual&MediaTypes=Video&ImageTypeLimit=1&recursive=true&Filters=IsResumable&Fields=PrimaryImageAspectRatio%2COverview%2CItemCounts%2CDisplayPreferencesId%2CChildCount&sortOrder=Descending&Limit=50&format=json
03-11 23:45:56.430  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items/Latest?ParentId=79a2726d3c50e769a8af1e4184e4fccf&Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&Limit=50&GroupItems=true&format=json
03-11 23:45:56.431  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Shows/NextUp?Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&UserId=2d1757004ca24c6a8a54241cf2bb2390&Limit=50&ImageTypeLimit=1&format=json
03-11 23:45:56.432  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items/Latest?ParentId=f137a2dd21bbc1b99aa5c0f6bf02a805&Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&Limit=50&GroupItems=true&format=json
03-11 23:45:56.441  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d1757004ca24c6a8a54241cf2bb2390/Items/Latest?ParentId=a656b907eb3a73532e40e44b968d0225&Fields=PrimaryImageAspectRatio%2COverview%2CChildCount&Limit=50&GroupItems=true&format=json
03-11 23:45:56.598  5767  5979 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=61e0900c-7d93-40b6-9805-c4e2c06d69fa, tags={ org.jellyfin.androidtv.integration.LeanbackChannelWorker } ]
03-11 23:45:56.895  5767  5785 I lyfin.androidt: Background young concurrent copying GC freed 34289(1716KB) AllocSpace objects, 8(180KB) LOS objects, 7% free, 18MB/19MB, paused 349us total 202.212ms
03-11 23:45:56.905  5767  5787 W System  : A resource failed to call end. 
03-11 23:45:56.906  5767  5787 I chatty  : uid=10064(org.jellyfin.androidtv) FinalizerDaemon identical 10 lines
03-11 23:45:57.092  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 135 lines
03-11 23:45:57.150  5767 28551 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:45:57.157  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 6 lines
03-11 23:45:57.546  5767  5785 I chatty  : uid=10064(org.jellyfin.androidtv) HeapTaskDaemon expire 2 lines
03-11 23:46:01.052  5767  6800 I chatty  : uid=10064(org.jellyfin.androidtv) AsyncTask #1 expire 1 line
03-11 23:46:04.762  5767 28568 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:06.944  5767 28577 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:07.154  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 271 lines
03-11 23:46:07.866  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 15 lines
03-11 23:46:09.259  5767 28584 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:10.987  5767  5785 I chatty  : uid=10064(org.jellyfin.androidtv) HeapTaskDaemon expire 2 lines
03-11 23:46:11.767  5767 28588 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:12.012  5767  5767 W ViewRootImpl[FullDetailsActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24313109, downTime=24313083, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:12.013  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 23:46:12.013  5767  5767 W ViewRootImpl[FullDetailsActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24313109, downTime=24313083, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:13.594  5767  5842 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-13 expire 1 line
03-11 23:46:13.978  5767 28594 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:16.230  5767  5841 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-12 expire 1 line
03-11 23:46:16.346  5767 28598 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:18.073  5767  5844 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-15 expire 1 line
03-11 23:46:18.509  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 319 lines
03-11 23:46:18.724  5767 28603 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:18.769  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 16 lines
03-11 23:46:18.804  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Dropping event due to no window focus: KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=24319563, downTime=24319563, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:18.816  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24319634, downTime=24319634, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:18.819  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 23:46:18.819  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24319634, downTime=24319634, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:20.052  5767  5787 I chatty  : uid=10064(org.jellyfin.androidtv) FinalizerDaemon expire 19 lines
03-11 23:46:20.743  5767  5843 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-14 expire 1 line
03-11 23:46:20.962  5767 28608 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:22.638  5767  5785 I chatty  : uid=10064(org.jellyfin.androidtv) HeapTaskDaemon expire 1 line
03-11 23:46:22.649  5767  5767 W ViewRootImpl[UserViewActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24322451, downTime=24321821, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:22.655  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 23:46:22.655  5767  5767 W ViewRootImpl[UserViewActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24322451, downTime=24321821, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:23.024  5767  5842 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-13 expire 1 line
03-11 23:46:25.377  5767 28620 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:26.219  5767  5841 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-12 expire 1 line
03-11 23:46:27.509  5767 28628 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:29.020  5767  5844 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-15 expire 1 line
03-11 23:46:29.451  5767  5843 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-14 expire 1 line
03-11 23:46:29.975  5767 28635 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:30.100  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Dropping event due to no window focus: KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=24331222, downTime=24331222, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:30.100  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24331278, downTime=24331222, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:30.102  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 23:46:30.102  5767  5767 W ViewRootImpl[PlaybackOverlayActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24331278, downTime=24331222, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:46:31.493  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 10 lines
03-11 23:46:31.728  5767  5787 I chatty  : uid=10064(org.jellyfin.androidtv) FinalizerDaemon expire 8 lines
03-11 23:46:32.316  5767 28647 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:34.022  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 56 lines
03-11 23:46:34.182  5767  5842 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-13 expire 1 line
03-11 23:46:34.881  5767 28553 I chatty  : uid=10064(org.jellyfin.androidtv) DefaultDispatch expire 3 lines
03-11 23:46:34.914  5767 28358 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:46:35.078  5767 27569 I chatty  : uid=10064(org.jellyfin.androidtv) DefaultDispatch expire 2 lines
03-11 23:46:35.164  5767 28552 I chatty  : uid=10064(org.jellyfin.androidtv) DefaultDispatch expire 1 line
03-11 23:46:35.539  5767  5806 I chatty  : uid=10064(org.jellyfin.androidtv) WM.task-1 expire 1 line
03-11 23:46:35.842  5767  5785 I chatty  : uid=10064(org.jellyfin.androidtv) HeapTaskDaemon expire 2 lines
03-11 23:46:36.608  5767  5841 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-12 expire 1 line
03-11 23:46:40.193  5767  5844 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-15 expire 1 line
03-11 23:46:41.015  5767  5843 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-14 expire 1 line
03-11 23:47:56.708  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 27 lines
03-11 23:48:06.788  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 11 lines
03-11 23:48:06.868  5767  6800 I chatty  : uid=10064(org.jellyfin.androidtv) AsyncTask #1 expire 1 line
03-11 23:48:35.500  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 127 lines
03-11 23:48:36.154  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 3 lines
03-11 23:48:37.766  5767 28691 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:48:38.329  5767  5785 I chatty  : uid=10064(org.jellyfin.androidtv) HeapTaskDaemon expire 1 line
03-11 23:48:38.354  5767  5787 I chatty  : uid=10064(org.jellyfin.androidtv) FinalizerDaemon expire 6 lines
03-11 23:48:39.912  5767 28695 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:48:40.245  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 101 lines
03-11 23:48:40.624  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 5 lines
03-11 23:48:42.062  5767 28700 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:48:43.494  5767  5785 I chatty  : uid=10064(org.jellyfin.androidtv) HeapTaskDaemon expire 1 line
03-11 23:48:44.292  5767 28706 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:48:44.616  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 17 lines
03-11 23:48:44.801  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 1 line
03-11 23:48:45.490  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 6 lines
03-11 23:48:46.353  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 1 line
03-11 23:48:46.361  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 3 lines
03-11 23:48:46.457  5767  5841 I chatty  : uid=10064(org.jellyfin.androidtv) Thread-12 expire 1 line
03-11 23:48:46.513  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 6 lines
03-11 23:48:46.567  5767  5830 I chatty  : uid=10064(org.jellyfin.androidtv) RenderThread expire 1 line
03-11 23:48:46.574  5767 28711 I chatty  : uid=10064(org.jellyfin.androidtv) expire 1 line
03-11 23:48:46.584  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) expire 45 lines
03-11 23:48:47.745  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 959(67KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 16MB/33MB, paused 124us total 217.619ms
03-11 23:48:47.965  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 7(32KB) AllocSpace objects, 0(0B) LOS objects, 50% free, 16MB/33MB, paused 129us total 218.559ms
03-11 23:48:48.181  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 15(32KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 16MB/33MB, paused 123us total 215.355ms
03-11 23:48:48.400  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 21(31KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 16MB/33MB, paused 127us total 216.943ms
03-11 23:48:48.641  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 22(48KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 17MB/35MB, paused 123us total 241.163ms
03-11 23:48:48.649  5767  5842 D Volley  : [232] NetworkUtility.logSlowRequests: HTTP response for request=<[ ] http://calor:8096/Sessions/Playing/Stopped 0x5a0d231 NORMAL 414> [lifetime=6523], [size=0], [rc=204], [retryCount=0]
03-11 23:48:48.651  5767  5830 W libEGL  : EGLNativeWindowType 0xc76a9808 disconnect failed
03-11 23:48:48.667  5767  5830 I OpenGLRenderer: Davey! duration=1376ms; Flags=1, IntendedVsync=24468479845219, Vsync=24468479845219, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=24468480189791, AnimationStart=24468480236041, PerformTraversalsStart=24468480692750, DrawStart=24469851243042, SyncQueued=24469851403084, SyncStart=24469852940834, IssueDrawCommandsStart=24469853116667, SwapBuffers=24469856262000, FrameCompleted=24469858311459, DequeueBufferDuration=663000, QueueBufferDuration=710000, 
03-11 23:48:48.682  5767  5767 I Choreographer: Skipped 82 frames!  The application may be doing too much work on its main thread.
03-11 23:48:48.687  5767  5830 I OpenGLRenderer: Davey! duration=1381ms; Flags=0, IntendedVsync=24468496528552, Vsync=24469864561858, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=24469872829125, AnimationStart=24469872856834, PerformTraversalsStart=24469873624125, DrawStart=24469873728542, SyncQueued=24469873916209, SyncStart=24469874440667, IssueDrawCommandsStart=24469874722667, SwapBuffers=24469877098542, FrameCompleted=24469878375042, DequeueBufferDuration=847000, QueueBufferDuration=464000, 
03-11 23:48:48.689  5767  5767 D FullDetailsActivity: current time 1647035328689 last playback event time 1647035326513 last refresh time 1647035326587
03-11 23:48:48.690  5767  5767 D VideoManager: Player is buffering
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal: Playback error
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:   com.google.android.exoplayer2.ExoPlaybackException: Source error
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:628)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:598)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:103)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:214)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:67)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:   Caused by: com.google.android.exoplayer2.ParserException: Missing CodecPrivate for codec A_AAC
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor$Track.getCodecPrivate(MatroskaExtractor.java:2460)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor$Track.initializeOutput(MatroskaExtractor.java:2045)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor.endMasterElement(MatroskaExtractor.java:770)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor$InnerEbmlProcessor.endMasterElement(MatroskaExtractor.java:1879)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.DefaultEbmlReader.read(DefaultEbmlReader.java:84)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor.read(MatroskaExtractor.java:480)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.source.BundledExtractorsAdapter.read(BundledExtractorsAdapter.java:127)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:1046)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:409)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-11 23:48:48.691  5767 28719 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:919)
03-11 23:48:48.691  5767  5767 D FullDetailsActivity: current time 1647035328691 last playback event time 1647035326513 last refresh time 1647035326587
03-11 23:48:48.693  5767  5830 I OpenGLRenderer: Davey! duration=1383ms; Flags=0, IntendedVsync=24468496528552, Vsync=24469864561858, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=24469872829125, AnimationStart=24469872856834, PerformTraversalsStart=24469873624125, DrawStart=24469874957125, SyncQueued=24469875043834, SyncStart=24469879049084, IssueDrawCommandsStart=24469879513417, SwapBuffers=24469882619417, FrameCompleted=24469884120375, DequeueBufferDuration=1219000, QueueBufferDuration=611000, 
03-11 23:48:48.699  5767  5830 W libEGL  : EGLNativeWindowType 0xe9587248 disconnect failed
03-11 23:48:48.706  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Sessions/Playing/Stopped
03-11 23:48:48.726  5767  5767 W ViewRootImpl[FullDetailsActivity]: Dropping event due to no window focus: KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=24469099, downTime=24469099, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:48:48.727  5767  5767 W ViewRootImpl[FullDetailsActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24469915, downTime=24469099, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:48:48.730  5767  5767 I chatty  : uid=10064(org.jellyfin.androidtv) identical 7 lines
03-11 23:48:48.730  5767  5767 W ViewRootImpl[FullDetailsActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x28, repeatCount=0, eventTime=24469915, downTime=24469099, deviceId=-1, source=0x2000001, displayId=-1 }
03-11 23:48:48.731  5767  5767 E VideoManager: ***** Got error from player
03-11 23:48:48.735  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Users/2d175700-4ca2-4c6a-8a54-241cf2bb2390/Items/2779e7117091cdae1116b7a374c4f42f?format=json
03-11 23:48:48.753  5767  5767 D BackgroundServiceFragment: Restoring active backgrounds
03-11 23:48:48.754  5767  5767 D BackgroundServiceFragment: Restoring background drawable
03-11 23:48:48.758  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Users/2d175700-4ca2-4c6a-8a54-241cf2bb2390/Items/2779e7117091cdae1116b7a374c4f42f?format=json
03-11 23:48:48.760  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/LiveTv/Channels/2779e7117091cdae1116b7a374c4f42f?userId=2d175700-4ca2-4c6a-8a54-241cf2bb2390&format=json
03-11 23:48:48.836  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/LiveTv/Channels/2779e7117091cdae1116b7a374c4f42f?userId=2d175700-4ca2-4c6a-8a54-241cf2bb2390&format=json
03-11 23:48:48.856  5767  5767 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@f126c05
03-11 23:48:48.867  5767  5767 I AppThemeCallbacks: Applying theme: DARK
03-11 23:48:48.889  5767  5767 E ThemeUtils: View class org.jellyfin.androidtv.ui.shared.StrokeTextView is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
03-11 23:48:48.911  5767  5767 I DefaultRenderersFactory: Loaded FfmpegAudioRenderer.
03-11 23:48:48.914  5767  5767 I ExoPlayerImpl: Init 9be59d1 [ExoPlayerLib/2.16.1] [sti6130d350, Dynalink TV Box, askey, 29]
03-11 23:48:48.923  5767  5767 D PlaybackController: Play called from state: IDLE with pos: 0 and sub index: null
03-11 23:48:48.925  5767  5767 D CustomPlaybackOverlayFragment: current server version [10.7.7] is < [10.8.0]
03-11 23:48:48.926  5767  5767 D PlaybackController: HLS is disabled
03-11 23:48:48.928  5767  5767 I PlaybackController: *** Using extended Exoplayer profile options
03-11 23:48:48.929  5767  5767 D PlaybackController: Max bitrate is: 120000000
03-11 23:48:48.929  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/LiveTv/Channels/2779e7117091cdae1116b7a374c4f42f?userId=2d1757004ca24c6a8a54241cf2bb2390&format=json
03-11 23:48:48.930  5767  5767 I PlaybackController: Using internal player for Live TV
03-11 23:48:48.931  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Items/2779e7117091cdae1116b7a374c4f42f/PlaybackInfo?format=json
03-11 23:48:48.934  5767  5767 W AudioManager: Use of stream types is deprecated for operations other than volume control
03-11 23:48:48.934  5767  5767 W AudioManager: See the documentation of requestAudioFocus() for what to use instead with android.media.AudioAttributes to qualify your playback use case
03-11 23:48:48.947  5767  5767 D CustomPlaybackOverlayFragment: Stopping!
03-11 23:48:48.948  5767  5767 D CustomPlaybackOverlayFragment: this fragment belongs to the current session, ending it
03-11 23:48:48.948  5767  5767 D PlaybackController: stop called at 0
03-11 23:48:49.099  5767  5767 I jellyfin-apiclient: ReportPlaybackStopped: Item 2779e7117091cdae1116b7a374c4f42f, Ticks: 0
03-11 23:48:49.099  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Sessions/Playing/Stopped
03-11 23:48:49.101  5767  5767 I ExoPlayerImpl: Release fb6afb4 [ExoPlayerLib/2.16.1] [sti6130d350, Dynalink TV Box, askey, 29] [goog.exo.core, goog.exo.ui, goog.exo.decoder, goog.exo.exoplayer, goog.exo.datasource, goog.exo.extractor, goog.exo.ffmpeg]
03-11 23:48:49.169  5767  5767 D MediaManager: Removed event listener.  Total listeners: 2
03-11 23:48:49.179  5767  5830 W libEGL  : EGLNativeWindowType 0xc3120f08 disconnect failed
03-11 23:48:49.182  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/LiveTv/Channels/2779e7117091cdae1116b7a374c4f42f?userId=2d1757004ca24c6a8a54241cf2bb2390&format=json
03-11 23:48:49.189  5767  5767 I TvManager: *** Channels already loaded - returning 8 channels
03-11 23:48:49.189  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Items/2779e7117091cdae1116b7a374c4f42f/PlaybackInfo?format=json
03-11 23:48:49.190  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/LiveStreams/Open?format=json
03-11 23:48:49.240  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/LiveStreams/Open?format=json
03-11 23:48:49.249  5767  5767 I VideoManager: Network buffer set to 600
03-11 23:48:49.252  5767  5767 D VideoManager: Surface attached
03-11 23:48:49.252  5767  5767 D PlaybackController: default audio index set to -1
03-11 23:48:49.252  5767  5767 D PlaybackController: default sub index set to -1 remote default null
03-11 23:48:49.253  5767  5767 I PlaybackController: Playing back in native mode.
03-11 23:48:49.257  5767  5767 I TvManager: *** Channels already loaded - returning 8 channels
03-11 23:48:49.258  5767  5767 I VideoManager: Video path set to: http://calor:8096/videos/2779e711-7091-cdae-1116-b7a374c4f42f/stream.mkv?DeviceId=84e5c9791211b140b426578901f8cfc2dea7067c&MediaSourceId=48bd876278ac7013551e4907ef6f2844&VideoCodec=h264&AudioCodec=aac,mp3&AudioStreamIndex=-1&VideoBitrate=119808000&AudioBitrate=192000&PlaySessionId=4eecac84b63447a5bbfa6693ea19c562&api_key=d9de48e8e52f4c24a5d2fc6287fa7d8b&LiveStreamId=a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_48bd876278ac7013551e4907ef6f2844&CopyTimestamps=true&RequireAvc=false&h264-profile=high,main,baseline,constrainedbaseline&h264-level=51&h264-maxrefframes=4&aac-audiochannels=6&mp3-audiochannels=6&TranscodeReasons=ContainerNotSupported,VideoCodecNotSupported,AudioCodecNotSupported
03-11 23:48:49.260  5767  5767 D VideoManager: Caught player timeline change - reason: PLAYLIST_CHANGED
03-11 23:48:49.261  5767  5767 D VideoManager: Player is buffering
03-11 23:48:49.262  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Sessions/Playing
03-11 23:48:49.263  5767  5767 I ReportingHelper: Playback of Lietuviški svingeriai (LNK) started.
03-11 23:48:49.274  5767  5767 D VideoManager: Caught player timeline change - reason: SOURCE_UPDATE
03-11 23:48:49.294  5767  5767 I jellyfin-apiclient: Response received from: http://calor:8096/Sessions/Playing
03-11 23:48:49.327  5767  5767 I HomeFragment: Updating audio queue in HomeFragment (onResume)
03-11 23:48:49.328  5767  5767 D BackgroundServiceFragment: Restoring active backgrounds
03-11 23:48:49.329  5767  5767 D BackgroundServiceFragment: Restoring background drawable
03-11 23:48:49.332  5767  5767 D BackgroundServiceFragment: Restoring active backgrounds
03-11 23:48:49.333  5767  5767 D BackgroundServiceFragment: Restoring background drawable
03-11 23:48:49.839  5767  5830 W libEGL  : EGLNativeWindowType 0xc4a25748 disconnect failed
03-11 23:48:50.073  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 45046(3488KB) AllocSpace objects, 3(124KB) LOS objects, 49% free, 16MB/32MB, paused 167us total 222.398ms
03-11 23:48:50.084  5767  5787 W System  : A resource failed to call end. 
03-11 23:48:50.084  5767  5787 I chatty  : uid=10064(org.jellyfin.androidtv) FinalizerDaemon identical 7 lines
03-11 23:48:50.084  5767  5787 W System  : A resource failed to call end. 
03-11 23:48:50.283  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 10951(537KB) AllocSpace objects, 0(0B) LOS objects, 50% free, 15MB/31MB, paused 134us total 198.481ms
03-11 23:48:50.487  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 661(52KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 130us total 202.669ms
03-11 23:48:50.687  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 15(32KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 121us total 199.209ms
03-11 23:48:50.834  5767  5843 D Volley  : [233] NetworkUtility.logSlowRequests: HTTP response for request=<[ ] http://calor:8096/Sessions/Playing/Stopped 0x5a0d231 NORMAL 421> [lifetime=6521], [size=0], [rc=204], [retryCount=0]
03-11 23:48:50.891  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 23(47KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 131us total 202.828ms
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal: Playback error
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:   com.google.android.exoplayer2.ExoPlaybackException: Source error
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:628)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:598)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:103)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:214)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:67)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:   Caused by: com.google.android.exoplayer2.ParserException: Missing CodecPrivate for codec A_AAC
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor$Track.getCodecPrivate(MatroskaExtractor.java:2460)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor$Track.initializeOutput(MatroskaExtractor.java:2045)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor.endMasterElement(MatroskaExtractor.java:770)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor$InnerEbmlProcessor.endMasterElement(MatroskaExtractor.java:1879)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.DefaultEbmlReader.read(DefaultEbmlReader.java:84)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.extractor.mkv.MatroskaExtractor.read(MatroskaExtractor.java:480)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.source.BundledExtractorsAdapter.read(BundledExtractorsAdapter.java:127)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:1046)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:409)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-11 23:48:51.072  5767 28724 E ExoPlayerImplInternal:       at java.lang.Thread.run(Thread.java:919)
03-11 23:48:51.088  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 790(171KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 17MB/35MB, paused 121us total 195.946ms
03-11 23:48:51.089  5767  5767 D CustomPlaybackOverlayFragment: Stopping!
03-11 23:48:51.090  5767  5767 D CustomPlaybackOverlayFragment: this fragment belongs to the current session, ending it
03-11 23:48:51.090  5767  5767 D PlaybackController: stop called at 0
03-11 23:48:51.241  5767  5767 I jellyfin-apiclient: ReportPlaybackStopped: Item 2779e7117091cdae1116b7a374c4f42f, Ticks: 0
03-11 23:48:51.241  5767  5767 D jellyfin-apiclient: Adding request to queue: http://calor:8096/Sessions/Playing/Stopped
03-11 23:48:51.244  5767  5767 I ExoPlayerImpl: Release 9be59d1 [ExoPlayerLib/2.16.1] [sti6130d350, Dynalink TV Box, askey, 29] [goog.exo.core, goog.exo.ui, goog.exo.decoder, goog.exo.exoplayer, goog.exo.datasource, goog.exo.extractor, goog.exo.ffmpeg]
03-11 23:48:51.270  5767  5830 W libEGL  : EGLNativeWindowType 0xc4a23088 disconnect failed
03-11 23:48:51.285  5767  5767 D MediaManager: Removed event listener.  Total listeners: 1
03-11 23:48:51.504  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 5772(1888KB) AllocSpace objects, 3(124KB) LOS objects, 49% free, 15MB/31MB, paused 154us total 211.746ms
03-11 23:48:51.702  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 145(36KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 124us total 196.323ms
03-11 23:48:51.917  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 7(32KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 120us total 214.352ms
03-11 23:48:52.134  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 20(48KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 125us total 216.349ms
03-11 23:48:52.342  5767  5767 I lyfin.androidt: Explicit concurrent copying GC freed 22(47KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 15MB/31MB, paused 181us total 206.443ms
```


### FFmpeg logs

```shell
/videos/35207466-0605-0f7f-185b-eb9024bf76e2/stream.mkv

{"Protocol":1,"Id":"6ac9c89694263d534393fb2960257d84","Path":"http://127.0.0.1:8368/_/https://www.lrt.lt/mediateka/tiesiogiai/lrt-lituanica","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"hls","Size":97280,"Name":null,"IsRemote":false,"ETag":null,"RunTimeTicks":null,"ReadAtNativeFramerate":false,"IgnoreDts":true,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":true,"RequiresOpening":true,"OpenToken":null,"RequiresClosing":true,"LiveStreamId":"a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_6ac9c89694263d534393fb2960257d84","BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":null,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"h264","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/90000","CodecTimeBase":null,"Title":null,"VideoRange":"SDR","localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"720p H264 SDR","NalLengthSize":"0","IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":8000000,"BitDepth":8,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":720,"Width":1280,"AverageFrameRate":25,"RealFrameRate":25,"Profile":"Main","Type":1,"AspectRatio":"16:9","Index":-1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p","Level":31,"IsAnamorphic":false},{"Codec":"aac","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/90000","CodecTimeBase":null,"Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"AAC - Stereo","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"stereo","BitRate":192000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":2,"SampleRate":48000,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":"LC","Type":0,"AspectRatio":null,"Index":-1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":8192000,"Timestamp":null,"RequiredHttpHeaders":{"User-Agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.85 Safari/537.36"},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":3000,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/run/current-system/sw/bin/ffmpeg -analyzeduration 3000000 -user_agent "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.85 Safari/537.36" -fflags +igndts+genpts -i "http://127.0.0.1:8368/_/https://www.lrt.lt/mediateka/tiesiogiai/lrt-lituanica" -sn -codec:v:0 copy -fflags +genpts -vsync -1 -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 copy -y "/var/lib/jellyfin/transcodes/fd2c3c6453d38a69b4f0ed8123a183da.mkv"


ffmpeg version 4.4.1 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.3.0 (GCC)
  configuration: --disable-static --prefix=/nix/store/zcyffhfyml0zcq7xpxnvgbfsp1vxr46n-jellyfin-ffmpeg --target_os=linux --arch=x86_64 --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --disable-small --enable-runtime-cpudetect --disable-lto --enable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --enable-ffmpeg --enable-ffplay --enable-ffprobe --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avresample --enable-avutil --enable-postproc --enable-swresample --enable-swscale --enable-doc --disable-htmlpages --enable-manpages --disable-podpages --disable-txtpages --enable-bzlib --enable-libcelt --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --enable-libfribidi --enable-libgme --enable-gnutls --enable-libgsm --enable-ladspa --enable-libmp3lame --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdc1394 --enable-libdrm --enable-iconv --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmysofa --enable-libopus --enable-librsvg --enable-libsrt --enable-libssh --enable-libtheora --enable-libv4l2 --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvmaf --enable-libvpx --enable-libwebp --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-libxcb-shape --enable-lzma --enable-nvenc --enable-openal --enable-libopencore-amrnb --disable-opengl --enable-libopenjpeg --disable-openssl --enable-libpulse --disable-librav1e --enable-libsvtav1 --disable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --enable-libvidstab --enable-libvo-amrwbenc --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-zlib --enable-vulkan --enable-libglslang --enable-libsmbclient --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[hls @ 0x1c99880] Skip ('#EXT-X-VERSION:3')
[hls @ 0x1c99880] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966933.ts' for reading
[hls @ 0x1c99880] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966934.ts' for reading
Input #0, hls, from 'http://127.0.0.1:8368/_/https://www.lrt.lt/mediateka/tiesiogiai/lrt-lituanica':
  Duration: N/A, start: 33427.594222, bitrate: N/A
  Program 0 
    Metadata:
      variant_bitrate : 0
  Stream #0:0: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuv420p, 1280x720, 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 0
  Stream #0:1: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 0
Output #0, matroska, to '/var/lib/jellyfin/transcodes/fd2c3c6453d38a69b4f0ed8123a183da.mkv':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Video: h264 (Main) (H264 / 0x34363248), yuv420p, 1280x720, q=2-31, 25 fps, 25 tbr, 1k tbn, 90k tbc
  Stream #0:1: Audio: aac (LC) ([255][0][0][0] / 0x00FF), 48000 Hz, stereo, fltp
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
frame=    0 fps=0.0 q=-1.0 size=       1kB time=00:00:00.00 bitrate=N/A speed=N/A    
[https @ 0x377c000] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966935.ts' for reading
[hls @ 0x1c99880] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/index.m3u8' for reading
[hls @ 0x1c99880] Skip ('#EXT-X-VERSION:3')
[https @ 0x377c000] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966936.ts' for reading
frame=  156 fps= 78 q=-1.0 size=     512kB time=00:00:06.31 bitrate= 663.9kbits/s speed=3.15x    
[https @ 0x3773400] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/index.m3u8' for reading
[hls @ 0x1c99880] Skip ('#EXT-X-VERSION:3')
[https @ 0x377c000] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966937.ts' for reading
frame=  206 fps= 51 q=-1.0 Lsize=    1136kB time=00:00:08.31 bitrate=1119.0kbits/s speed=2.06x    
video:936kB audio:197kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.247115%
```


### Media info of the file

```shell
ffprobe version 4.4.1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 10.3.0 (GCC)
  configuration: --disable-static --prefix=/nix/store/zcyffhfyml0zcq7xpxnvgbfsp1vxr46n-jellyfin-ffmpeg --target_os=linux --arch=x86_64 --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --disable-small --enable-runtime-cpudetect --disable-lto --enable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --enable-ffmpeg --enable-ffplay --enable-ffprobe --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avresample --enable-avutil --enable-postproc --enable-swresample --enable-swscale --enable-doc --disable-htmlpages --enable-manpages --disable-podpages --disable-txtpages --enable-bzlib --enable-libcelt --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --enable-libfribidi --enable-libgme --enable-gnutls --enable-libgsm --enable-ladspa --enable-libmp3lame --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdc1394 --enable-libdrm --enable-iconv --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmysofa --enable-libopus --enable-librsvg --enable-libsrt --enable-libssh --enable-libtheora --enable-libv4l2 --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvmaf --enable-libvpx --enable-libwebp --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-libxcb-shape --enable-lzma --enable-nvenc --enable-openal --enable-libopencore-amrnb --disable-opengl --enable-libopenjpeg --disable-openssl --enable-libpulse --disable-librav1e --enable-libsvtav1 --disable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --enable-libvidstab --enable-libvo-amrwbenc --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-zlib --enable-vulkan --enable-libglslang --enable-libsmbclient --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[hls @ 0x1d47d40] Skip ('#EXT-X-VERSION:3')
[hls @ 0x1d47d40] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966983.ts' for reading
[hls @ 0x1d47d40] Opening 'https://lituanica-dvr.lrt.lt/lrt-ndvr/hls/lituanica_720p/15966984.ts' for reading
Input #0, hls, from 'http://127.0.0.1:8368/_/https://www.lrt.lt/mediateka/tiesiogiai/lrt-lituanica':
  Duration: N/A, start: 33527.114222, bitrate: N/A
  Program 0
    Metadata:
      variant_bitrate : 0
  Stream #0:0: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuv420p, 1280x720, 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 0
  Stream #0:1: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 0
```


### Application version

0.13.0-beta.5

### Device information

Dynalink 4K Box

### Android version

Android 10

### Jellyfin server version

10.7.7

--- a/app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java
+++ b/app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java
@@ -175,7 +175,7 @@ public class PlaybackController {
     }
 
     public boolean hasInitializedVideoManager() {
-        return mVideoManager != null && mVideoManager.isInitialized() ? true : false;
+        return mVideoManager != null && mVideoManager.isInitialized();
     }
 
     public MediaSourceInfo getCurrentMediaSource() {
@@ -270,7 +270,6 @@ public class PlaybackController {
             Timber.i("Player error encountered - retrying");
             stop();
             play(mCurrentPosition);
-
         } else {
             mPlaybackState = PlaybackState.ERROR;
             if (mFragment != null) {
@@ -322,7 +321,7 @@ public class PlaybackController {
 
     @TargetApi(23)
     private void setRefreshRate(MediaStream videoStream) {
-        if (videoStream == null) {
+        if (videoStream == null || mFragment == null) {
             Timber.e("Null video stream attempting to set refresh rate");
             return;
         }
@@ -1360,18 +1359,19 @@ public class PlaybackController {
 
             @Override
             public void onEvent() {
-                if (mFragment == null) return;
+                if (mFragment == null) {
+                    playerErrorEncountered();
+                    return;
+                }
+
                 if (isLiveTv && directStreamLiveTv) {
                     Utils.showToast(mFragment.getContext(), mFragment.getString(R.string.msg_error_live_stream));
                     directStreamLiveTv = false;
-                    PlaybackHelper.retrieveAndPlay(getCurrentlyPlayingItem().getId(), false, mFragment.getContext());
-                    mFragment.finish();
                 } else {
                     String msg = mFragment.getString(R.string.video_error_unknown_error);
                     Timber.e("Playback error - %s", msg);
-                    playerErrorEncountered();
                 }
-
+                playerErrorEncountered();
             }
         });
 

--- a/app/src/main/java/org/jellyfin/androidtv/util/profile/ExoPlayerProfile.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/util/profile/ExoPlayerProfile.kt
@@ -55,34 +55,51 @@ class ExoPlayerProfile(
 	init {
 		name = "AndroidTV-ExoPlayer"
 
-		if (isHlsSupported) {
-			transcodingProfiles = arrayOf(
-					// TS video profile
-					TranscodingProfile().apply {
-						type = DlnaProfileType.Video
-						context = EncodingContext.Streaming
-						container = ContainerTypes.TS
-						videoCodec = buildList {
-							if (deviceHevcCodecProfile.ContainsCodec(CodecTypes.HEVC, ContainerTypes.TS)) add(CodecTypes.HEVC)
-							add(CodecTypes.H264)
-						}.joinToString(",")
-						audioCodec = buildList {
-							if (isAC3Enabled) add(CodecTypes.AC3)
-							add(CodecTypes.AAC)
-							add(CodecTypes.MP3)
-						}.joinToString(",")
-						protocol = "hls"
-						copyTimestamps = false
-					},
-					// MP3 audio profile
-					TranscodingProfile().apply {
-						type = DlnaProfileType.Audio
-						context = EncodingContext.Streaming
-						container = CodecTypes.MP3
-						audioCodec = CodecTypes.MP3
-					}
-			)
-		}
+		transcodingProfiles = arrayOf(
+			when (isHlsSupported) {
+				// TS video profile
+				true -> TranscodingProfile().apply {
+					type = DlnaProfileType.Video
+					context = EncodingContext.Streaming
+					container = ContainerTypes.TS
+					videoCodec = buildList {
+						if (deviceHevcCodecProfile.ContainsCodec(CodecTypes.HEVC, ContainerTypes.TS)) add(CodecTypes.HEVC)
+						add(CodecTypes.H264)
+					}.joinToString(",")
+					audioCodec = buildList {
+						if (isAC3Enabled) add(CodecTypes.AC3)
+						add(CodecTypes.AAC)
+						add(CodecTypes.MP3)
+					}.joinToString(",")
+					protocol = "hls"
+					copyTimestamps = false
+				}
+				// MKV video profile
+				else -> TranscodingProfile().apply {
+					type = DlnaProfileType.Video
+					context = EncodingContext.Streaming
+					container = ContainerTypes.MKV
+					videoCodec = buildList {
+						if (deviceHevcCodecProfile.ContainsCodec(CodecTypes.HEVC, ContainerTypes.MKV)) add(CodecTypes.HEVC)
+						add(CodecTypes.H264)
+					}.joinToString(",")
+					audioCodec = buildList {
+						if (isAC3Enabled) add(CodecTypes.AC3)
+						add(CodecTypes.AAC)
+						add(CodecTypes.MP3)
+					}.joinToString(",")
+					protocol = "http"
+					copyTimestamps = true
+				}
+			},
+			// MP3 audio profile
+			TranscodingProfile().apply {
+				type = DlnaProfileType.Audio
+				context = EncodingContext.Streaming
+				container = CodecTypes.MP3
+				audioCodec = CodecTypes.MP3
+			}
+		)
 
 		directPlayProfiles = buildList {
 			// Video direct play

